tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -360,15 +344,71 @@ var _ = Describe("QinQ", Ordered, Label(tsparams.LabelQinQTestCases), ContinueOn
+					[]string{srIovNetworkDot1AD, srIovNetworkDot1AD, nadMasterBond0},
+					[]string{tsparams.ServerIPv4IPAddress, tsparams.ServerIPv6IPAddress})
+
+				serverPod := createServerTestPod(serverNameDot1ad, workerNodeList[0].Definition.Name, testCmdBond0,
+					serverAnnotation)
+
+				By("Define and create a 802.1AD client container")
+				clientAnnotation := pod.StaticIPBondAnnotationWithInterface("nadcvlan100", "bond0.100",
+					[]string{srIovNetworkDot1AD, srIovNetworkDot1AD, nadMasterBond0},
+					[]string{tsparams.ClientIPv4IPAddress, tsparams.ClientIPv6IPAddress})
+				clientPod := createClientTestPod(clientNameDot1ad, workerNodeList[0].Definition.Name, clientAnnotation)
+
+				By("Validate IPv4 and IPv6 connectivity between the containers over the qinq tunnel.")
+				err = cmd.ICMPConnectivityCheck(serverPod, clientIPAddressesNet2, intBond0)
+				Expect(err).ToNot(HaveOccurred(),
+					"Failed to ping the client container over the 802.1AD connection")
+
+				By("Validate IPv4 and IPv6 tcp traffic and dot1ad encapsulation from the client to server")
+				validateTCPTraffic(clientPod, intBond0, serverIPAddressesNet2)
+
+				By("Validate that the TCP traffic is double tagged")
+				readAndValidateTCPDump(tcpDumpContainer, tcpDumpReadFileCMD, tcpDumpDot1ADOutput)
+			})
 	})
 
 	Context("802.1Q", func() {
 		BeforeAll(func() {
-			By("Define and create sriov-network with 802.1q S-VLAN")
-			vlan, err := strconv.Atoi(NetConfig.VLAN)
-			Expect(err).ToNot(HaveOccurred(), "Failed to convert VLAN value")
-			defineAndCreateSrIovNetworkWithQinQ(srIovNetworkDot1Q, srIovPolicyResNameNetDevice, dot1q,
-				uint16(vlan))
+			By("Verify Sriov Polices are created.")
+			policyList, err := sriov.ListPolicy(APIClient, NetConfig.SriovOperatorNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Unable to pull Policy list")
+
+			if len(policyList) != 2 {
+				By("Define and create a sriovNetworkPolicies Netdevice and VfioPCI")
+				defineCreateSriovNetPolices(srIovPolicyNetDevice, srIovPolicyVfioPci, srIovPolicyResNameNetDevice,
+					srIovPolicyResNameVfioPci, srIovInterfacesUnderTest[0], sriovDeviceID)
+
+				By("Define and create a sriovNetworks")
+				defineAndCreateSriovNetworks(srIovNetworkPromiscuous, srIovNetworkDot1AD, srIovNetworkDot1Q,
+					srIovPolicyResNameNetDevice)
+
+				defineAndCreateNAD(nadCVLAN100, nadCVLAN101, nadMasterBond0, intNet1)
+
+				promiscVFCommand = fmt.Sprintf("ethtool --set-priv-flags %s vf-true-promisc-support on",
+					srIovInterfacesUnderTest[0])
+				if sriovDeviceID == mlxDevice {
+					promiscVFCommand = fmt.Sprintf("ip link set %s promisc on",
+						srIovInterfacesUnderTest[0])
+				}
+
+				By(fmt.Sprintf("Enable VF promiscuous support on %s", srIovInterfacesUnderTest[0]))
+				output, err := cmd.RunCommandOnHostNetworkPod(workerNodeList[0].Definition.Name,
+					NetConfig.SriovOperatorNamespace, promiscVFCommand)
+				Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to run command on node %s", output))
```

Comment:
This is repeats several time in the tests. Please create a func

Reply:
created func